alexandria/core/filters.py

```diff
@@ -164,6 +187,19 @@ class FileFilterSet(FilterSet):
     document_metainfo = JSONValueFilter(field_name="document__metainfo")
     active_group = ActiveGroupFilter()
     files = BaseCSVFilter(field_name="pk", lookup_expr="in")
+    content_search = FullTextSearchFilter(field_name="content_vector")
+    latest = BooleanFilter(method="filter_latest")
+
+    def filter_latest(self, qs, name, value):
+        if not value:
+            return qs
+
+        latest_files = models.File.objects.filter(
+            document=OuterRef("document_id")
+        ).order_by("-created_at")
+        latest_file_of_documents = qs.filter(pk=Subquery(latest_files.values("pk")[:1]))
```

Comment:
Can you try and see what the actual SQL query comes out of this? I'm afraid this might not perform very well.

If neccessary, I think we should add an `is_latest` field to `File` that's updated when new versions are added..

But then again, maybe we need to iterate on this anyway: We want the latest version with a *match* in the search, which may not be in the latest actual version of the file...

Reply:
In a previous iteration of the filter I was generating the subquery with
```py
qs.filter(document=OuterRef("document_id")).order_by("-created_at")
```
This was filtering for the latest matching file.

The current query filters for the latest file of the document.
Django generated output:
```sql
SELECT
    "alexandria_core_file"."created_at",
    "alexandria_core_file"."created_by_user",
    "alexandria_core_file"."created_by_group",
    "alexandria_core_file"."modified_at",
    "alexandria_core_file"."modified_by_user",
    "alexandria_core_file"."modified_by_group",
    "alexandria_core_file"."metainfo",
    "alexandria_core_file"."id",
    "alexandria_core_file"."variant",
    "alexandria_core_file"."original_id",
    "alexandria_core_file"."name",
    "alexandria_core_file"."document_id",
    "alexandria_core_file"."checksum",
    "alexandria_core_file"."encryption_status",
    "alexandria_core_file"."content_vector",
    "alexandria_core_file"."language",
    "alexandria_core_file"."content",
    "alexandria_core_file"."mime_type",
    "alexandria_core_file"."size"
FROM
    "alexandria_core_file"
WHERE
    (
        "alexandria_core_file"."variant" = 'original'
        AND "alexandria_core_file"."id" = (
            SELECT
                U0."id"
            FROM
                "alexandria_core_file" U0
            WHERE
                U0."document_id" = "alexandria_core_file"."document_id"
            ORDER BY
                U0."created_at" DESC
            LIMIT
                1
        )
    )
ORDER BY
    "alexandria_core_file"."created_at" DESC;
```

Reply:
OK, subquery looks good to me, but it's missing the `variant` filter. And thumbnails tend to be newer than the originals, so this will likely break your assumptions

Reply:
Done